### DISCUSSION TITLE
TLS/SSL not working, can't connect with any mail program

### DISCUSSION THREAD

**Discussion by vfxdee** [modoboa/modoboa#2682] [Help]
Hi.  I just installed modoboa.  Helm is going bankrupt, so I have to make my own mail server and modoboa seemed to be the easiest to setup.  Can anyone tell me what I did wrong?  I used the install instructions on the readthedocs page https://modoboa.readthedocs.io/en/latest/installation.html
It went through all the steps automatically with the wonderful setup script. It worked wonderfully except that tls/ssl/start aren't working.  When I try to sign in with my email client, I get this:
![Screenshot_Email](https://user-images.githubusercontent.com/10745817/200504500-68c53c91-4884-4cbb-8647-9e1cf270e54d.png)

When I try to login on my phone, it gives me the error that it doesn't understand the TLS response.

Also, it doesn't allow me to load the web ui securely because it says no SSL.  
It also says it can't connect to the seive server. It should be installed by the setup script, correct? Or do I need to do that myself?
![Screenshot from 2022-11-08 00-02-56](https://user-images.githubusercontent.com/10745817/200508251-719f2b57-60b0-4a0c-92c8-5c45522c6569.png)
I confirmed that the sieve plugin is installed, so it should be working.  I'm guessing the ssl certificate problem has an effect on this, too?
I have no idea what to put in for the autoconfig and autodiscover dns records because I don't know what the hostname it's wanting is.

I'm sorry for being so new to this.  Does anyone know what I need to do to fix any of this?

when I look at /var/log/mail.err, it shows:
`Nov  8 06:37:56 localhost dovecot: imap-login: Error: Failed to initialize SSL server context: Can't load SSL certificate (ssl_cert setting): The certificate is empty: user=<>, rip=108.180.29.73, lip=139.177.195.44, session=<XZDMx+/sZrJstB1J>
Nov  8 06:45:03 localhost postfix/postscreen[33414]: error: open /etc/postfix/postscreen_spf_whitelist.cidr: No such file or directory
Nov  8 06:46:18 localhost dovecot: lmtp(35893): Error: SSL context initialization failed, disabling SSL: Can't load SSL certificate (ssl_cert setting): The certificate is empty
Nov  8 06:48:34 localhost dovecot: managesieve-login: Error: Failed to initialize SSL server context: Can't load SSL certificate (ssl_cert setting): The certificate is empty: user=<>, rip=127.0.0.1, lip=127.0.0.1, secured, session=<cjrM7e/sNq9/AAAB>
Nov  8 07:05:59 localhost postfix/postscreen[44470]: error: open /etc/postfix/postscreen_spf_whitelist.cidr: No such file or directory
Nov  8 07:08:58 localhost dovecot: imap-login: Error: Failed to initialize SSL server context: Can't load SSL certificate (ssl_cert setting): The certificate is empty: user=<>, rip=108.180.29.73, lip=139.177.195.44, session=<z9S8NvDsApFstB1J>
`

It seems to want an ssl certificate that wasn't generated or is empty, but I'm not sure where to put it or what exact arguments to give it.
Should I have run certbot somewhere? Was there a step somewhere that could've failed in the setup script and now I need to rerun it?

----Edit----

I reset my linode I had this on and reinstalled and now it doesn't allow me to access the webUI at all. Everything else shows that it's working, though I'm not sure the ssl is working again. It shows that it should be working, but all connections are refused. ufw is disabled, so it shouldn't be denying anything

**Comment by tonioo** [comment]
What kind of certificate did you generate during installation?

**Reply by vfxdee** [reply]
I used letsencrypt to generate the certificates.  I just used the setup script but changed the config to do letsencrypt. I can log in and send email through the web interface, but I can't use imap because its tls encryption isn't working. It shows
`Nov  9 15:58:44 mail dovecot: imap-login: Error: Failed to initialize SSL server context: Can't load SSL certificate: There is no valid PEM certificate.`

**Reply by tonioo** [reply]
Can you check dovecot configuration? And what's inside /etc/letsencrypt/live folder?

**Comment by vfxdee** [comment]
A discord community member, Brett (GodOfLinux#2553), fixed it. He said the include_try seems to break things. He also said he'd be happy to walk you through it in dm.
